src/main.py

```diff
@@ -24,12 +28,68 @@
 # Plot label distribution
 plt.figure()
 plt.bar(label_distribution.keys(), label_distribution.values(), 0.8)
-for x, y in enumerate(label_distribution.values()):
-    plt.text(x, y, str(y), horizontalalignment='center')
+for a, b in enumerate(label_distribution.values()):
+    plt.text(a, b, str(b), horizontalalignment='center')
 plt.show()
 
-"""Task 2 - Naive Bayes"""
+"""Task 2 - Naive Bayes Classifier"""
 multiNB = MultinomialNB()
 multiNB.fit(X_train, y_train)
-y_pred = multiNB.predict(X_test)
-pprint.pprint(metrics.classification_report(y_test, y_pred, digits=4, output_dict=True))
+multiNB_pred = multiNB.predict(X_test)
+
+"""Task 2 - Base-DT"""
+baseDT = DecisionTreeClassifier(criterion='entropy')
+baseDT.fit(X_train, y_train)
+baseDT_pred = baseDT.predict(X_test)
+
+"""Task 2 - Best-DT"""
+bestDT = DecisionTreeClassifier()
+bestDT.fit(X_train, y_train)
+bestDT_pred = bestDT.predict(X_test)
+
+"""Task 3 - Generate output"""
+labels = sorted(list(set(y)))
+score_list = ['precision', 'recall', 'f1-score']
+models = [('NaiveBayes', multiNB_pred), ('BaseDT', baseDT_pred), ('BestDT', bestDT_pred)]
+
+for (name, y_pred) in models:
```

Comment:
Can you place this loop in a function called `generate_output` in `utils` with `labels`, `score_list`, and `models` as parameters and call it here

Reply:
I would have to import numpy, confusion_matrix & classification_report in utils then, or just import numpy and move the calls for confusion_matrix and classification_report outside of the loop, but then it looks repetitive. What would you prefer?

Reply:
Take a look at the changes, let me know what you think

Reply:
looks good
You can merge it